### PR TITLE
feat(EMS-2399-2400): No PDF - Bump application version, About goods or services enhancements

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR:
  - Creates a new application version in the API so that:
    - No PDF has a different application version to MVP and some fields default to `null` instead of true/false.
  - Updates the "About goods or services" form so that:
  - A new "is the final destination known" question is included.
  - The existing "final destination" field is only required if the final destination is known.
  - The design is updated so that the "final destination" field is an autocomplete search input, instead of a select drop down.

## Resolution :heavy_check_mark:
- Update `completeAndSubmitAboutGoodsOrServicesForm` cypress command to have a condition and be in an object structure.
- Create new `isVisible` and `isNotVisible` assertions in the `checkAutocompleteInput` command, since we now have a conditional autocomplete input.
- Update content strings and error messages.
- Add/update E2E test coverage for:
  - Rendering and submitting the About goods or services form, with the final destination being known and unknown.
  - Changing the "is the final destination known" answer from "yes" to "no".
- Create new API application version 2 (current "no PDF" phase).
- Add new `FINAL_DESTINATION_KNOWN` field ID.
- Update the task list to pass and consume a `finalDestinationKnown` flag. If this is provided, "final destination" is a required to have a completed "policy" task list group.
- Update the export contract section's `mapAndSave` function to consume a countries array.
- Create new `mapSubmittedData` function to map the final destination country.
  - Ensured that if the "is the final destination known" question is changed from "yes" to "no", the originally submitted final destination is removed.
- Update About goods or services form validation rules.
- Update the About goods or services nunjucks template to render yes/no radios and a conditional autocomplete input.

## Miscellaneous :heavy_plus_sign:
- Minor documentation improvements.
- Fixed some typos.

### CODE REVIEW

File: e2e-tests/insurance/cypress/e2e/journeys/policy/about-goods-or-services/about-goods-or-services-final-destination-not-known.spec.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  },
+} = INSURANCE_FIELD_IDS;
+
+const task = taskList.prepareApplication.tasks.policy;
+
+const finalDestinationField = countryInput.field(FINAL_DESTINATION);
+
+const baseUrl = Cypress.config('baseUrl');
+
+context('Insurance - Policy - About goods or services page - final destination not known - As an exporter, I want to enter the details of the export contract', () => {
```
Comment (Zainzzkk) [inline e2e-tests/insurance/cypress/e2e/journeys/policy/about-goods-or-services/about-goods-or-services-final-destination-not-known.spec.js:34]: capital F in Final?

File: e2e-tests/insurance/cypress/e2e/journeys/policy/about-goods-or-services/about-goods-or-services.spec.js [hunk 2]
```diff
@@ -35,15 +38,17 @@ const {
...
 } = INSURANCE_FIELD_IDS;
 
 const task = taskList.prepareApplication.tasks.policy;
 
+const finalDestinationField = countryInput.field(FINAL_DESTINATION);
+
 const baseUrl = Cypress.config('baseUrl');
 
-context('Insurance - Policy - About goods or services page - As an exporter, I want to enter the details of the export contract', () => {
+context('Insurance - Policy - About goods or services page - final destination not known - As an exporter, I want to enter the details of the export contract', () => {
```
Comment (Zainzzkk) [inline e2e-tests/insurance/cypress/e2e/journeys/policy/about-goods-or-services/about-goods-or-services.spec.js:51]: same here?

File: e2e-tests/insurance/cypress/e2e/journeys/policy/about-goods-or-services/validation/about-goods-or-services-validation.spec.js [hunk 3]
```diff
@@ -65,14 +67,36 @@ context('Insurance - Policy - About goods or services page - form validation', (
...
   });
+
+  describe(`when ${FINAL_DESTINATION_KNOWN} is 'yes', but ${FINAL_DESTINATION} is not provided`, () => {
+    it(`should render a ${FINAL_DESTINATION} validation error`, () => {
+      cy.navigateToUrl(url);
+
+      cy.keyboardInput(aboutGoodsOrServicesPage[DESCRIPTION].textarea(), application.EXPORT_CONTRACT[DESCRIPTION]);
+      yesRadio().input().click();
+
+      submitButton().click();
```
Comment (Zainzzkk) [inline e2e-tests/insurance/cypress/e2e/journeys/policy/about-goods-or-services/validation/about-goods-or-services-validation.spec.js:88]: is there a command for the above?

File: src/ui/server/controllers/insurance/policy/about-goods-or-services/index.test.ts [hunk 4]
```diff
@@ -214,7 +226,10 @@ describe('controllers/insurance/policy/about-goods-or-services', () => {
 
         expect(mapAndSave.exportContract).toHaveBeenCalledTimes(1);
 
-        expect(mapAndSave.exportContract).toHaveBeenCalledWith(payload, res.locals.application);
+        const expectedValidationErrors = false;
+        // const expectedCountries = mapCountries(mockCountries, payload[FINAL_DESTINATION]);
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/policy/about-goods-or-services/index.test.ts:230]: remove commented out code?

File: e2e-tests/insurance/cypress/e2e/journeys/policy/about-goods-or-services/validation/about-goods-or-services-validation.spec.js (hunk 3 above)
Comment (ttbarnes) [reply e2e-tests/insurance/cypress/e2e/journeys/policy/about-goods-or-services/validation/about-goods-or-services-validation.spec.js:88]: Good shout, i've updated `completeAndSubmitAboutGoodsOrServicesForm` to have a `includeFinalDestination` flag. Therefore can use this here and declare the flag as false.

**Review by Zainzzkk — APPROVED** [review]

**Pull request merged**